packages/build-info/src/package-managers/detect-package-manager.ts

```diff
@@ -17,8 +17,8 @@ export type PkgManagerFields = {
   installCommand: string
   /** The package managers run command prefix */
   runCommand: string
-  /** The lock file a package manager is using */
-  lockFile: string
+  /** The lock files a package manager is using */
+  lockFiles: string[]
```

Comment:
We could potentially opt to support both fields for a short period of time while calling the old one deprecated (maybe raise a warning message in the logs?). That way we can understand the scope of affected systems before we look to drop the field.

Reply:
There is limited number of things using `@netlify/build-info` directly, so I think we can skip on interim period - this package is not meant for public consumption per-se as it's quite Netlify specific, so I don't think it warrants extra effort to have a ~deprecation period. Instead maybe we just have a canary with this change that we adjust consumers of this package before we merge and publish this change so we can minimize "package upgrade downtime"

Reply:
Published ~canary https://github.com/netlify/build/releases/tag/build-info-v8.0.0-pre-20250103.0.0 for testing (yikes with `preid` with double `.0` there :D I definitely held https://github.com/netlify/build/blob/main/.github/workflows/pre-release.yml wrong hehe )